src/main/java/io/cdap/plugin/http/source/common/http/HttpClient.java

```
@@ -59,29 +65,63 @@ public HttpClient(BaseHttpSourceConfig config) {
     } else {
       requestBody = null;
     }
+
+    if (config.getServiceAccountEnabled()) { // Service account authent used
+      try {
+        serviceAccountTokenHandler = new ServiceAccountTokenHandler(
+                config.getServiceAccountScope(),
+                config.getServiceAccountPrivateKeyJson()
+        );
+      } catch (Exception e) {
+        LOG.error("Error while getting Service Account Access token " + e);
```

We shouldn't ignore exception here.